src/cursor/cur_layered.c

```diff
@@ -966,6 +966,39 @@ __clayered_put(WT_SESSION_IMPL *session, WT_CURSOR_LAYERED *clayered, const WT_I
     return (0);
 }
 
+/*
+ * __clayered_modify_int --
+ *     Put an modiy into the desired tree.
+ */
+static WT_INLINE int
+__clayered_modify_int(WT_SESSION_IMPL *session, WT_CURSOR_LAYERED *clayered, const WT_ITEM *key,
+  WT_MODIFY *entries, int nentries)
+{
+    WT_CURSOR *c;
+
+    /*
+     * Clear the existing cursor position. Don't clear the primary cursor: we're about to use it
+     * anyway.
+     */
+    WT_RET(__clayered_reset_cursors(clayered, true));
+
+    WT_RET(__layered_modify_check(session));
```

Comment:
We don't need to call this a second time - I think the other (top-level) call is the right place for this.